### PR TITLE
Restructure code, fix different GPUs conflict error

### PR DESCRIPTION
**Pull request by tannyle289**
- Restructure code, fix different GPUs conflict error
- Fix build failed in previous release

### CODE REVIEW

**Comment by cedricve** [comment]
Thanks @tannyle289 as discussed we should move the models also into the project, and no longer depend on setting the `MODEL_NAME` in the environment variables, but solely depend on the concept of a project. Models belong to a project, and will be hard coded in the project.

File: projects/base_project.py [hunk 1]
```diff
@@ -34,11 +42,43 @@ def class_mapping(self, results1, results2):
...
+        # initialise the yolo model, additionally use the device parameter to specify the device to run the model on.
+        self.device = 'cuda' if torch.cuda.is_available() else 'cpu'
+        _cur_dir = pdirname(pabspath(__file__))
+        model_dir = pjoin(_cur_dir, f'../models')
+        model_dir = pabspath(model_dir)  # normalise the link
+
+        if not self._var.MODEL_NAME:
+            raise ModuleNotFoundError('Model not found!')
+
+        model = YOLO(pjoin(model_dir, self._var.MODEL_NAME)).to(self.device)
```
Comment (cedricve) [inline projects/base_project.py]: Models should go into project class, not in base project.

**Review by cedricve — CHANGES_REQUESTED** [review]

File: .env [hunk 2]
```diff
@@ -5,7 +5,7 @@ MODEL_NAME_2="your_second_model.pt"
 MODEL_ALLOWED_CLASSES="0"
 MODEL_2_ALLOWED_CLASSES="0"
```
Comment (cedricve) [inline .env]: we no longer need to have the model settings, we use project instead (which under the hood implements the models)

**Review by cedricve — CHANGES_REQUESTED** [review]

File: .env (hunk 2 above)
Comment (tannyle289) [reply .env]: Refactored

File: projects/base_project.py (hunk 1 above)
Comment (tannyle289) [reply projects/base_project.py]: Refactored

**Comment by tannyle289** [comment]
Now we can adapt with multiple models, for every project it will read models from project_config.yaml file regardless of the size (but at least 1 model)
Pic below as an example for 3 models 
image

**Pull request merged**